test/validation/api/ipsec/ipsec.c

```diff
@@ -370,6 +458,48 @@ void ipsec_sa_param_fill(odp_ipsec_sa_param_t *param,
 	param->lifetime.hard_limit.packets = 10000 * 1000;
 }
 
+static void ipsec_status_event_handle(odp_event_t ev_status,
+				      odp_ipsec_sa_t sa,
+				      enum ipsec_test_sa_expiry sa_expiry)
+{
+	odp_ipsec_status_t status = {
+		.id = 0,
+		.sa = ODP_IPSEC_SA_INVALID,
+		.result = 0,
+		.warn.all = 0,
+	};
+
+	CU_ASSERT_FATAL(ODP_EVENT_INVALID != ev_status);
+	CU_ASSERT_EQUAL(1, odp_event_is_valid(ev_status));
+	CU_ASSERT_EQUAL(ODP_EVENT_IPSEC_STATUS, odp_event_type(ev_status));
+
+	CU_ASSERT_EQUAL(0, odp_ipsec_status(&status, ev_status));
+	CU_ASSERT_EQUAL(ODP_IPSEC_STATUS_WARN, status.id);
+	CU_ASSERT_EQUAL(sa, status.sa);
+	CU_ASSERT_EQUAL(0, status.result);
+
+	if (IPSEC_TEST_EXPIRY_SOFT_PKT == sa_expiry) {
+		CU_ASSERT_EQUAL(1, status.warn.soft_exp_packets);
+		sa_expiry_notified = true;
+	} else if (IPSEC_TEST_EXPIRY_SOFT_BYTE == sa_expiry) {
+		CU_ASSERT_EQUAL(1, status.warn.soft_exp_bytes);
+		sa_expiry_notified = true;
+	}
+
+	odp_event_free(ev_status);
+}
+
+void ipsec_status_event_get(odp_ipsec_sa_t sa,
+			    enum ipsec_test_sa_expiry sa_expiry)
+{
+	uint64_t wait_time = sa_expiry_notified ? 0 : STATUS_EVENT_WAIT_TIME;
```

Comment:
The zero wait time has the potential issue that if events do come, they may not get received here but be left in the scheduler and get received at a later point.

Reply:
Ack. Only observation with same timeout for all the events was that the test was taking slightly longer time to run. Keeping the same timeout for all the cases.